src/js/jsx/shared/Dialog.jsx

```diff
@@ -124,11 +132,10 @@ define(function (require, exports, module) {
                 id = this.props.id;
 
             if (this.state.open) {
+                this._target = null;
                 flux.actions.dialog.closeDialog(id);
             } else if (!this.props.disabled) {
-                this.setState({
-                    target: event.target
-                });
+                this._target = event.target;
```

Comment:
I looked into that, but I don't think that's the intended use of that callback? In any case, I could not convince myself that that was safe. If you can point me to docs that say otherwise, I would be interested.